mongo/cursor_test.go

```diff
@@ -102,37 +102,57 @@ func TestCursor(t *testing.T) {
 	t.Run("TestAll", func(t *testing.T) {
 		t.Run("errors if argument is not pointer to slice", func(t *testing.T) {
 			cursor, err := newCursor(newTestBatchCursor(1, 5), nil, nil)
-			assert.Nil(t, err, "newCursor error: %v", err)
+			assert.NoError(t, err, "newCursor error: %v", err)
 			err = cursor.All(context.Background(), []bson.D{})
-			assert.NotNil(t, err, "expected error, got nil")
+			assert.Error(t, err, "expected error, got nil")
 		})
 
 		t.Run("fills slice with all documents", func(t *testing.T) {
 			cursor, err := newCursor(newTestBatchCursor(1, 5), nil, nil)
-			assert.Nil(t, err, "newCursor error: %v", err)
+			assert.NoError(t, err, "newCursor error: %v", err)
 
 			var docs []bson.D
 			err = cursor.All(context.Background(), &docs)
-			assert.Nil(t, err, "All error: %v", err)
-			assert.Equal(t, 5, len(docs), "expected 5 docs, got %v", len(docs))
+			assert.NoError(t, err, "All error: %v", err)
+			assert.Len(t, docs, 5, "expected 5 docs, got %v", len(docs))
 
 			for index, doc := range docs {
 				expected := bson.D{{"foo", int32(index)}}
 				assert.Equal(t, expected, doc, "expected doc %v, got %v", expected, doc)
 			}
 		})
 
+		t.Run("nil slice", func(t *testing.T) {
+			cursor, err := newCursor(newTestBatchCursor(0, 0), nil, nil)
+			assert.NoError(t, err, "newCursor error: %v", err)
+
+			var docs []bson.D
+			err = cursor.All(context.Background(), &docs)
+			assert.NoError(t, err, "All error: %v", err)
+			assert.Nil(t, docs, "expected nil docs")
+		})
+
+		t.Run("empty slice", func(t *testing.T) {
+			cursor, err := newCursor(newTestBatchCursor(0, 0), nil, nil)
+			assert.NoError(t, err, "newCursor error: %v", err)
+
+			docs := []bson.D{{{"foo", "bar"}}, {{"hello", "world"}, {"pi", 3.14159}}}
```

Comment:
Should the `docs` here be empty? I.e., `docs := []bson.D{}`. Should We add a third test for overwritten?